tests/common/plugins/custom_markers/__init__.py

```diff
@@ -89,3 +101,60 @@ def check_device_type(item):
+        3.3 Specified level is in between two defined levels - go to next lower level
+    4. Specified level matches one of the defined levels
+    '''
+    specified_level = item.config.getoption("--completeness_level")
+    if not specified_level: # Case 1
+        logging.info("Completeness level not set during test execution. Setting to default level: {}".format(str(CompletenessLevel.basic)))
+        specified_level = CompletenessLevel.basic # - every testcase should run BASIC by default
+    else:
+        specified_level = specified_level.lower()
+        if specified_level not in CompletenessLevel._member_names_:
+            specified_level = CompletenessLevel.basic
+            warnings.warn("Unidentified completeness level specified. Specified: {}. Allowed: {}".format(str(CompletenessLevel(specified_level)), \
+                str(CompletenessLevel._member_names_)))
+            logging.info("Unidentified completeness level specified. Setting to default level: {}".format(CompletenessLevel.debug))
+        else:
+            specified_level = CompletenessLevel[specified_level]
+    # The closest marker is used here so that the module or class level
+    # marker will be overrided by case level marker
+    defined_levels = [mark.args for mark in item.iter_markers(name="supported_completeness_level")]
+    if defined_levels is None or len(defined_levels) == 0: # Case 2
+        logging.info("Test has no defined levels. Continue without test completeness checks")
+        return
+    defined_levels = defined_levels[0] # The nearest mark overides others
+
+    logging.info("Setting test completeness level. Specified: {}. Defined: {}".format(str(CompletenessLevel(specified_level)), str(defined_levels)))
+
+    if specified_level not in defined_levels:
+        if specified_level > max(defined_levels): # Case 3.1
+            completeness_level = max(defined_levels)
+        elif specified_level < min(defined_levels): # Case 3.2
+            completeness_level = min(defined_levels)
+        else: # Case 3.3
+            # Find the maximum defined level less than specified_level
+            lesser_defined_level_dist = sys.maxsize
+            for level in defined_levels:
+                if level <= specified_level and lesser_defined_level_dist > (specified_level - level):
+                    completeness_level = level
+                    lesser_defined_level_dist = lesser_defined_level_dist - level
+        logging.info("Specified level ({}) not found in defined levels. Setting level to {}".format(str(CompletenessLevel(specified_level)), str(completeness_level)))
+    else: # Case 4
+        completeness_level = specified_level
+        logging.info("Setting the completeness level to {}".format(str(CompletenessLevel(completeness_level))))
+
+    normalized_completeness_level = pytest.mark.completeness_level(completeness_level)
+    item.add_marker(normalized_completeness_level)
```

Comment:
This creates a new marker. As an example:

If a testcase will have following markers at the beginning of `pytest_runtest_setup`:
```
pytestmark = [
    pytest.mark.disable_loganalyzer,
    pytest.mark.topology('any'),
    pytest.mark.supported_completeness_level(CompletenessLevel.debug, CompletenessLevel.basic)
]

```
If the specified level (CLI arg) is `debug` then at the end of `pytest_runtest_setup`, markers for the testcase will look like:
```
pytestmark = [
    pytest.mark.disable_loganalyzer,
    pytest.mark.topology('any'),
    pytest.mark.supported_completeness_level(CompletenessLevel.debug, CompletenessLevel.basic)
    pytest.mark.completeness_level(CompletenessLevel.debug)
]
```

Reply:
Thanks Vaibhav! I didn't notice that you were creating a new marker

Reply:
Update - modified the code to use just one marker instead of two. The normalized completeness level is now added to the list of supported levels. I had originally used two markers merely for simplicity reason. On the contrary it seems to be creating confusion now.